modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,494 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'json'
+require 'uri'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::FtpServer
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Atlassian Confluence Widget Connector Macro Velocity Template Injection",
+      'Description'    => %q{
+        Widget Connector Macro is part of Atlassian Confluence Server and Data Center that
+        allows embed online videos, slideshows, photostreams and more directly into page. 
+        A _template parameter can be used to inject remote Java code into a Velocity template, 
+        and gain code execution. Authentication is unrequired to exploit this vulnerability. 
+        By default, Java payload will be used because it is cross-platform, but you can also 
+        specify which native payload you want (Linux or Windows).
+
+        Confluence before version 6.6.12, from version 6.7.0 before 6.12.3, from version 
+        6.13.0 before 6.13.3 and from version 6.14.0 before 6.14.2 are affected.
+
+        This vulnerability was originally discovered by Daniil Dmitriev
```

Comment:
Please credit Daniil in the `Author` section. Thank you.

Reply:
Done.